tests/gordo_components/model/test_model.py

```diff
@@ -203,62 +204,35 @@ def test_lookback_window_forecast_valueerror_during_transform(self):
             with self.assertRaises(ValueError):
                 model.fit(X)
 
-    def test_generate_window_lstm_ae_output(self):
+    def test_get_TimeseriesGenerator(self):
         # Check that right output is generated from generate_window (for KerasLSTMAutoEncoder)
         X = np.random.random(size=10).reshape(5, 2)
 
         lookback_window = 4
-        model = KerasLSTMAutoEncoder(
-            kind=lstm_autoencoder.lstm_model, lookback_window=lookback_window
+        gen = get_TimeseriesGenerator(
+            X, batch_size=1, lookback_window=lookback_window, lookahead=0
         )
-        gen = model.generate_window(X)
+        gen = (e for e in gen)
         gen_out_1 = next(gen)
         gen_out_2 = next(gen)
         self.assertEqual(gen_out_1[0].tolist(), X[0:4].reshape(1, 4, 2).tolist())
         self.assertEqual(gen_out_2[0].tolist(), X[1:5].reshape(1, 4, 2).tolist())
         self.assertEqual(gen_out_1[1].tolist(), X[3].reshape(1, 2).tolist())
         self.assertEqual(gen_out_2[1].tolist(), X[4].reshape(1, 2).tolist())
 
-        X = np.random.random(size=8).reshape(4, 2)
-        lookback_window = 2
-        model = KerasLSTMAutoEncoder(
-            kind=lstm_autoencoder.lstm_model, lookback_window=lookback_window
-        )
-        gen_no_y = model.generate_window(X, output_y=False)
-        gen_no_y_out_1 = next(gen_no_y)
-        gen_no_y_out_2 = next(gen_no_y)
-        gen_no_y_out_3 = next(gen_no_y)
-        self.assertEqual(gen_no_y_out_1.tolist(), X[0:2].reshape(1, 2, 2).tolist())
-        self.assertEqual(gen_no_y_out_2.tolist(), X[1:3].reshape(1, 2, 2).tolist())
-        self.assertEqual(gen_no_y_out_3.tolist(), X[2:4].reshape(1, 2, 2).tolist())
-
-    def test_generate_window_lstm_forecast_output(self):
-        # Check that right output is generated from generate_window (for KerasLSTMForecast)
-        X = np.random.random(size=(5, 2))
-
         lookback_window = 3
-        model = KerasLSTMForecast(
-            kind=lstm_autoencoder.lstm_model, lookback_window=lookback_window
+
+        gen = get_TimeseriesGenerator(
+            X, batch_size=1, lookback_window=lookback_window, lookahead=1
         )
-        gen = model.generate_window(X)
+        gen = (e for e in gen)
```

Comment:
So `TimeseriesGenerator` is not a generator? Otherwise, maybe adding `iter()` to the returned value is better, or?

Reply:
No, pretty confusing name given to it by Keras guys ....